comp/core/configsync/configsyncimpl/module.go

```diff
@@ -88,6 +100,19 @@ func newConfigSync(deps dependencies, agentIPCPort int, configRefreshIntervalSec
 		ctx:       ctx,
 	}
 
+	if deps.SyncParams.SyncOnInit {
```

Comment:
Why not rely on the lifecycle hooks ?

Reply:
Because lifecycle hooks are applied after instantiation but prior to the app running. I need this delay while we're instantiating because by the time the config object is injected to `traceconfig` the sync needs to have already happened. We can hopefully change this once we can learn how to propagate changes within the `otel-agent`, but that's a larger effort that will come later.